src/util.ts

```diff
@@ -844,3 +844,45 @@ export function makeMemoizedByteOffsetConverter(buffer: Buffer): (byteOffset: nu
 	};
 }
 
+export class TempFileProvider {
+	private constructor() {}
+	private static _instance: TempFileProvider;
+
+	static get instance() {
```

Comment:
Why not make this a static class? Then, we can skip the `instance()` method here.

Reply:
Very good point. I'll make this a static class 😃